appveyor.yml

```
@@ -157,6 +160,25 @@ test_script:
      $wc = New-Object 'System.Net.WebClient'
      $wc.UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)", $unitTestsXml)
      if($errorCode -ne 0) { $host.SetShouldExit($errorCode) }
+ - ps: |
+    if($env:APPVEYOR_PULL_REQUEST_NUMBER) {
+      $lintOutput = (Resolve-Path .\testOutput\lint\)
+      $lintSource = (Resolve-Path .\tests\lint\)
+      git fetch -q origin $env:APPVEYOR_REPO_BRANCH
+      $prDiff = "$lintOutput\prDiff.patch"
+      git diff -u FETCH_HEAD...HEAD > $prDiff

```

The problem with the current diff approach is that flake8 also parses the untouched lines in the diff and prints warnings about them. I think this might be fine at some point, but if we ever decide that linting must be ok before we allow a pull request to pass, it requires the author to touch lots of code that is part of the context rather than the actually changed code. And.. you know what's going to happen, people update the context, then new context is added to the diff which also results into warnings.

This can be fixed by getting less lines of context in the diff, e.g.

```suggestion
      git diff -U0 -u FETCH_HEAD...HEAD > $prDiff
```

This is pretty extreme though.

```suggestion
      git diff -U1 -u FETCH_HEAD...HEAD > $prDiff
```

This might be a nice intermediate choice.

---

Good point, I think U0 is the better choice. Start with the most minimal requirement. The intention is that linting should pass before the final review / merge. We may not be strict about this to start with, but its the goal.